lib/matplotlib/lines.py

```diff
@@ -801,8 +801,13 @@ def draw(self, renderer):
                 # subsample the markers if markevery is not None
                 markevery = self.get_markevery()
                 if markevery is not None:
+                    try:
+                        transform = self.axes.transAxes
+                    except AttributeError:
+                        # Typically in the case of a **figure** legend.
+                        transform = self.get_transform()
```

Comment:
I think this should just be set to None (or IdentityTransform)?  I don't know if we really support float markeverys for figure-level artists (this PR only requires support for list-of-indexes markeverys), so setting it to None to throw something (in the `isinstance(step, float)` case of _mark_every_path) may be better (well, it probably used to throw here, so it's not worse), or if we want to support that case as well (which may be worth noting as a new, if obscure, feature), I think the figure-level artist's get_transform() already includes everything that's needed so we may not need to redo the transform another time? (in which case IdentityTransform may be correct?)